CMakeLists.txt

```diff
@@ -20,6 +20,11 @@ set(OPENSSL_VERSION_NUMBER  0x11100000L)
 # Generates compile_commands.json file for clangd to parse.
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
+if ($ENV{XP_BUILD})
+    message("-- Building libsnowflakeclient for XP")
+    add_definitions(-DLIBSFCLI_FOR_XP)
+endif ()
+
```

Comment:
In the next PR i will provide a script which has the capability to set these environmental variables for XP build 
and also has the ability to pick up a particular commit and build.